pkg/collector/adapters/config_to_probe_test.go

```diff
@@ -0,0 +1,243 @@
+package adapters_test
+
+import (
+	"github.com/open-telemetry/opentelemetry-operator/pkg/collector/adapters"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestSimpleCase(t *testing.T) {
+	configStr := `extensions:
+  health_check:
+service:
+  extensions: [health_check]`
+
+	// prepare
+	config, err := adapters.ConfigFromString(configStr)
+	require.NoError(t, err)
+	require.NotEmpty(t, config)
+
+	// test
+	actualProbe, err := adapters.ConfigToContainerProbe(logger, config)
+	assert.NoError(t, err)
+	assert.Equal(t, "/", actualProbe.HTTPGet.Path)
+	assert.Equal(t, int32(13133), actualProbe.HTTPGet.Port.IntVal)
+	assert.Equal(t, "0.0.0.0", actualProbe.HTTPGet.Host)
+}
+
+func TestShouldUseCustomEndpointAndPath(t *testing.T) {
+	configStr := `extensions:
+  health_check:
+    endpoint: localhost:1234
+    path: /checkit
+service:
+  extensions: [health_check]`
+
+	// prepare
+	config, err := adapters.ConfigFromString(configStr)
+	require.NoError(t, err)
+	require.NotEmpty(t, config)
+
+	// test
+	actualProbe, err := adapters.ConfigToContainerProbe(logger, config)
+	assert.NoError(t, err)
+	assert.Equal(t, "/checkit", actualProbe.HTTPGet.Path)
+	assert.Equal(t, int32(1234), actualProbe.HTTPGet.Port.IntVal)
+	assert.Equal(t, "localhost", actualProbe.HTTPGet.Host)
+}
+
+func TestShouldUseCustomEndpointAndDefaultPath(t *testing.T) {
+	configStr := `extensions:
+  health_check:
+    endpoint: localhost:1234
```

Comment:
One case seems to be missing: `:1234`. Can you add it there as well?

Reply:
good call, thanks.